src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -1019,68 +1020,128 @@ int SketchObject::movePoint(int GeoId, PointPos PosId, const Base::Vector3d& toP
     return lastSolverStatus;
 }
 
+template <>

```

Comment:
These function doesn't depends of SketchObject, shouldn't they move to a SketchUtil.cpp (or whatever we name it)?

Reply:
Maybe. One step at a time. However, note that this is a very specific function taking a sketcher-specific function and getting the point based on geometry of the object (which is obtained through Part and ultimately OCCT). Such functionality has been part of SketchObject (with `Sketch` handling the functionality connected to planeGCS).

Reply:
+1 though this was what I attempted to do in https://github.com/FreeCAD/FreeCAD/pull/10721 and there was a roadblock which is that very annoyingly utils functions cannot be shared between GUI and APP.

To be shared you currently need to wrap them in a class object. And at this point then why not using the sketch object class...

Another alternative would be to have the same Util file in both App and Gui and keep them in sync.